libvips/colour/XYZ2CMYK.c

```diff
@@ -94,20 +94,64 @@ vips_XYZ2CMYK_build( VipsObject *object )
 	return( 0 );
 }
 
+#else
+
+typedef VipsColourCode VipsXYZ2CMYK;
+typedef VipsColourCodeClass VipsXYZ2CMYKClass;
+
+G_DEFINE_TYPE(VipsXYZ2CMYK, vips_XYZ2CMYK, VIPS_TYPE_COLOUR_CODE);
+
+void
+vips_XYZ2CMYK_line( VipsColour *colour, VipsPel *out, VipsPel **in, int width )
+{
+    float *p = (float *) in[0];
+    unsigned char *q = (unsigned char *) out;
+
+    int i;
+
+    for (i = 0; i < width; i++, p += 3, q += 4) {
+        float R, G, B;
+        int r, g, b, og;
+        int c, m, y, k;
+        
+        vips_col_XYZ2scRGB(p[0], p[1], p[2], &R, &G, &B);
+        vips_col_scRGB2sRGB_8(R, G, B, &r, &g, &b, &og);
+
+        if (r == 0 && g == 0 & b == 0) {
+            q[3] = 255;
+            continue;
```

Comment:
You'll need to set `q[0] / 1 / 2` as well. 

I don't like continue in loops except for errors, I think this would be clearer as an if / else, if it's still necessary.

Reply:
Yes I see, I did it because q[1..2] will have no impact on color rendering if q[3] == 255.
I will change that